cmd/operator-sdk/generate/csv.go

```diff
@@ -191,25 +182,28 @@ func validateVersion(version string) error {
 	return nil
 }
 
-func writeCRDsToDir(crdPaths []string, toDir string) error {
-	for _, p := range crdPaths {
-		b, err := ioutil.ReadFile(p)
-		if err != nil {
-			return err
-		}
-		typeMeta, err := k8sutil.GetTypeMetaFromBytes(b)
-		if err != nil {
-			return err
-		}
-		if typeMeta.Kind != "CustomResourceDefinition" {
-			continue
-		}
+// findCRDFileSet searches all directories and files in path for CRD manifests,
+// returning a map of paths to file contents.
+func findCRDFileSet(path string) (map[string][]byte, error) {
+	crdFileSet := map[string][]byte{}
+	info, err := os.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("crd's must be read from a directory. %s is a file", path)
```

Comment:
WDYT? E.g:
```suggestion
		return nil, fmt.Errorf("The path informed by the flag `--api-dir` is for a file and not a directory")
```

Reply:
`--api-dir` does not specify the CRD manifests location. That would be the path to the API types e.g `pkg/apis/...`.
But actually we don't even have an `--api-dir` flag.

Reply:
Just to clarifies the suggestion:
> "The path informed by the flag X is for a file and not a directory as expected")

Ps. `--api-dir` is just an example since I am advocating for above we have just flags and not a flag that we can pass a keymap. However, the suggestion with the keymap would something as: 

> "The path informed by the flag [X] and in the key [Y] with the value [Z] is a path for a file and not a directory as expected")
 
is it make more sense?